### PR TITLE
#118: Create device for aperture motors and add to snapshot plan

### PR DESCRIPTION
**Pull request by ndg63276**
Not sure how best to get values from beamlineParameters

### CODE REVIEW

File: src/artemis/snapshot_plan.py [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 
 RE = RunEngine({})
 
-def move_then_wait():
-	yield from abs_set(backlight.pos, Backlight.OUT, group='A')
+def prepare_for_snapshot():
+	yield from abs_set(backlight.pos, Backlight.IN, group='A')
+	# TODO get from beamlineParameters
+	miniap_y_ROBOT_LOAD = 31.40
+	if aperture.y.position > miniap_y_ROBOT_LOAD:
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/artemis/snapshot_plan.py:16]: Apologies, I realise I originally told you to do using `position` in the first place.

File: src/artemis/snapshot_plan.py [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+aperture.wait_for_connection()
 
 RE = RunEngine({})
 
-def move_then_wait():
-	yield from abs_set(backlight.pos, Backlight.OUT, group='A')
+def prepare_for_snapshot():
+	yield from abs_set(backlight.pos, Backlight.IN, group='A')
+	# TODO get from beamlineParameters
+	miniap_y_ROBOT_LOAD = 31.40
```
Comment (DominicOram) [inline src/artemis/snapshot_plan.py:15]: **nit**: I realise this is a bit temporary until we figure out how to do beamlineParameters but can we rename this to be more snake_case/descriptive. It's also not clear why we're using this position given that the robot isn't being used here at all. So maybe either rename to something like `aperture_y_out_of_beam` or a comment as to why we're referencing the robot.

Comment (DominicOram) [reply src/artemis/snapshot_plan.py:15]: I've added a comment to https://github.com/DiamondLightSource/python-artemis/issues/53 that we need to get `beamlineParams` working ASAP

File: src/artemis/devices/aperture.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from ophyd import Component, Device, EpicsMotor
+
+class Aperture(Device):
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/artemis/devices/aperture.py:3]: If you have no objections to this but don't want to do it as part of this I'll make a new issue, will be a good simple first issue for someone.

**Review by DominicOram — CHANGES_REQUESTED** [review]
Looks good, thanks. Some comments in code but mostly very minor.

**Comment by DominicOram** [comment]
I think we can tackle getting the beamlineParameters separately in https://github.com/DiamondLightSource/python-artemis/issues/53 and will update this accordingly once that's done.

File: src/artemis/devices/aperture.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+	x: EpicsMotor = Component(EpicsMotor, "-MO-MAPT-01:X")
+	y: EpicsMotor = Component(EpicsMotor, "-MO-MAPT-01:Y")
+	z: EpicsMotor = Component(EpicsMotor, "-MO-MAPT-01:Z")
```
[the remark this answers is not in the document]
Comment (ndg63276) [reply src/artemis/devices/aperture.py]: Done in aa8ab02073f724070f95ad7db21e08fdabbf10fb

File: src/artemis/devices/aperture.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ndg63276) [reply src/artemis/devices/aperture.py:3]: No objections, but I think make a new issue to have a generic bundle.

File: src/artemis/snapshot_plan.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndg63276) [reply src/artemis/snapshot_plan.py]: Done in https://github.com/DiamondLightSource/python-artemis/pull/121/commits/9dea9f1bcb57cf42618890373d763f7e08ccf6aa

File: src/artemis/snapshot_plan.py (hunk 2 above)
Comment (ndg63276) [reply src/artemis/snapshot_plan.py]: The name was pulled straight from beamlineParameters, but have updated it in https://github.com/DiamondLightSource/python-artemis/pull/121/commits/d3c387f99b968f6e50fc7b6926d5d76866a09247

File: src/artemis/devices/aperture.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/artemis/devices/aperture.py:3]: Done in https://github.com/DiamondLightSource/python-artemis/issues/130

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**